src/index.ts

```diff
@@ -304,15 +436,20 @@ export default class ImageTool {
    *                              {@link https://github.com/codex-team/editor.js/blob/master/types/tools/paste-events.d.ts}
    * @returns {void}
    */
-  async onPaste(event) {
+  async onPaste(event: CustomEvent): Promise<void> {
     switch (event.type) {
       case 'tag': {
         const image = event.detail.data;
 
         /** Images from PDF */
         if (/^blob:/.test(image.src)) {
           const response = await fetch(image.src);
-          const file = await response.blob();
+          /**
+           * the return value of response.blob() is Promise<blob>, which can't be used in the function uploadFile.
+           * So, it is converted to File, and then use the file in the uploadFile function.
+           */
+          const blob = await response.blob();
+          const file = new File([blob], 'image.jpg', { type: blob.type, lastModified: Date.now() });
```

Comment:
this changes the existing logic. Maybe change the `this.uploadFile(file);` argument type to Blob?